runbotics/runbotics-scheduler/src/scheduler-database/user/user.service.ts

```
@@ -363,4 +398,25 @@ export class UserService {
             relations: ['authorities'],
         });
     }
+
+    async updateLanguage(userId: number, langKey: string): Promise<User> {
+        const supportedLanguages = Object.values(Language);
+
+        if (!supportedLanguages.includes(langKey as SupportedLanguage)) {
+            throw new BadRequestException(
+                `Language '${langKey}' is not supported. Supported languages: ${supportedLanguages.join(', ')}`
+            );
+        }
+        const updateResult = await this.userRepository.update(userId, {
+            langKey,
+        });
+
+        if (updateResult.affected === 0) {
+            throw new BadRequestException(
+                'User not found or language not updated'
+            );
+        }
+
+        return await this.userRepository.findOne({ where: { id: userId } });
+    }
```

I have 2 small issues with that, but happy to discuss:
1. you first check is for language support but I think more important is if that user even exists. if it doesn't exist why even bother to check language? why not use findOneBy (throw not found here if !user), then validate the language and then just save method (which returns entity - no need for findBy and update in separate calls). "Language not updated" is not needed info in my opinion - user wanted that language and that language is saved.
2. you throw BadRequestException but you say in message that not found (NotFoundException, 404) and not updated (304, but nestjs doesn't have that so imo this should not be an error or just HttpException('Not Modified', HttpStatus.NOT_MODIFIED)